### PR TITLE
also consider $LMOD_CMD in bootstrap script

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
@verdurin: please test?

**Comment by verdurin** [comment]
@boegel: the bootstrapping works:

```
[[INFO]] Found module command '/usr/share/lmod/lmod/libexec/lmod' via $LMOD_CMD (Lmod), so using it.
```

and I'm able to load EB. However, when I run it there's an error:

```
[vagrant@localhost eb]$ eb -S SAMtools
== temporary log file in case of crash /tmp/eb-TxN07l/easybuild-0eecm7.log
== Searching (case-insensitive) for 'SAMtools' in /home/vagrant/.local/easybuild/software/EasyBuild/2.7.0/lib/python2.7/site-packages/easybuild_easyconfigs-2.7.0-py2.7.egg/easybuild/easyconfigs
CFGS1=/home/vagrant/.local/easybuild/software/EasyBuild/2.7.0/lib/python2.7/site-packages/easybuild_easyconfigs-2.7.0-py2.7.egg/easybuild/easyconfigs
 * $CFGS1/b/BSMAP/BSMAP-2.74_samtools-deps.patch
 * $CFGS1/s/SAMtools/SAMtools-0.1.18-goalf-1.1.0-no-OFED.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.18-goolf-1.4.10.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.18-ictce-4.0.6.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.18-ictce-5.3.0.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.19-foss-2015a.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.19-goolf-1.4.10.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.19-goolf-1.7.20.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.19-ictce-5.5.0.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.19-intel-2015a.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.19_Makefile-ncurses.patch
 * $CFGS1/s/SAMtools/SAMtools-0.1.20-goolf-1.7.20.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.20-intel-2015b.eb
 * $CFGS1/s/SAMtools/SAMtools-0.1.20_Makefile-ncurses.patch
 * $CFGS1/s/SAMtools/SAMtools-1.1-foss-2014b.eb
 * $CFGS1/s/SAMtools/SAMtools-1.1-goolf-1.4.10.eb
 * $CFGS1/s/SAMtools/SAMtools-1.1-intel-2014b.eb
 * $CFGS1/s/SAMtools/SAMtools-1.1-intel-2015a.eb
 * $CFGS1/s/SAMtools/SAMtools-1.1_Makefile.patch
 * $CFGS1/s/SAMtools/SAMtools-1.2-foss-2015a-HTSlib-1.2.1.eb
 * $CFGS1/s/SAMtools/SAMtools-1.2-foss-2015a.eb
 * $CFGS1/s/SAMtools/SAMtools-1.2-foss-2015b.eb
 * $CFGS1/s/SAMtools/SAMtools-1.2-goolf-1.4.10.eb
 * $CFGS1/s/SAMtools/SAMtools-1.2-goolf-1.7.20.eb
 * $CFGS1/s/SAMtools/SAMtools-1.2-intel-2015a-HTSlib-1.2.1.eb
 * $CFGS1/s/SAMtools/SAMtools-1.2-intel-2015b-HTSlib-1.2.1.eb
 * $CFGS1/s/SAMtools/SAMtools-1.2_extHTSlib_Makefile.patch
 * $CFGS1/s/SAMtools/SAMtools-1.3-foss-2015b.eb
 * $CFGS1/s/SAMtools/SAMtools-1.3-foss-2016a.eb
 * $CFGS1/s/SAMtools/SAMtools-1.3-intel-2016a.eb
 * $CFGS1/s/SAMtools/SAMtools-1.3_Makefile.patch
 * $CFGS1/s/SAMtools/SAMtools_Makefile-ncurses.patch
ERROR: EnvironmentModulesC modules tool can not be used, 'modulecmd' command is not available.
```

**Comment by boegel** [comment]
@verdurin Thanks for the confirmation.

The runtime problem you run into is because the default setting for the modules tool is to use `modulecmd` (i.e. the Tcl/C environment modules implementation).

I'm not sure if we want to change the default to something more dynamic...

For the bootstrap, it's fine, since it doesn't affect the end result: the installation and generated module file are exactly the same, regardless of the module tool being used.

However, if both `modulecmd` and `lmod` (or `$LMOD_CMD`) are available in the environment, I wouldn't want `eb` to pick one over the other randomly...

**Comment by boegel** [comment]
@wpoely86 please review?

File: easybuild/scripts/bootstrap_eb.py [hunk 1]
```diff
@@ -167,31 +167,40 @@ def check_module_command(tmpdir):
...
+    for modcmd, modtool in known_module_commands:
+        if check_cmd_help(modcmd):
             easybuild_modules_tool = modtool
             info("Found module command '%s' (%s), so using it." % (modcmd, modtool))
             break
-
-    if modtool is None:
+        elif modcmd == 'lmod':
+            # check value of $LMOD_CMD as fallback
+            modcmd = os.environ.get('LMOD_CMD')
```
Comment (wpoely86) [inline easybuild/scripts/bootstrap_eb.py]: I think `$LMOD_CMD` should get preference for `$PATH`. The Lmod source script only sets `$LMOD_CMD` not path.

Comment (boegel) [inline easybuild/scripts/bootstrap_eb.py]: That's not how the framework does it (when `Lmod` is used as value for `--modules-tool`)...

It's mostly for backwards compatibility: what if you have both `lmod` in your `$PATH`, and `$LMOD_CMD` defined to something else? Weird setup, sure...

Comment (wpoely86) [inline easybuild/scripts/bootstrap_eb.py]: well, then `$LMOD_CMD` wins. It's the way Lmod works. The whole `$PATH` thing should be a fallback. Unless you do it very explicitly yourself, `lmod` will not be in the `$PATH`.

Comment (boegel) [inline easybuild/scripts/bootstrap_eb.py]: Indeed, and since that's how people have been doing it for a while (making sure `lmod` is available), we need to take backwards compatibility into account...

In practice, it really shouldn't matter: if you don't have `lmod` in `$PATH`, it'll pick up `$LMOD_CMD`. So fine, right?

Comment (wpoely86) [inline easybuild/scripts/bootstrap_eb.py]: Well, I would pick first what Lmod will pick: `$LMOD_CMD`. Whatever you do with your `$PATH` doesn't really matter.

Comment (boegel) [inline easybuild/scripts/bootstrap_eb.py]: Well, ok, but that should be considered in a separate PR.

And this that's a change that breaks backward compatibility, I'd like to talk this over with a couple of the 'regulars' before it gets merged.

So, let's merge this PR, and change it to let `$LMOD_CMD` win consistently in both the bootstrap script and framework itself in a stand-alone PR.

**Comment by boegel** [comment]
@verdurin Thanks for testing
@wpoely86 Thanks for reviewing

Follow-up on giving `$LMOD_CMD` preference will be done in a separate PR; up for tackling that @wpoely86?
